lib/webdriverio.js

```diff
@@ -200,7 +200,7 @@ let WebdriverIO = function (args, modifier) {
         err.shotTaken = true
         return takeScreenshot(err)
             .catch((e) => logger.log('\tFailed to take screenshot on reject:', e))
-            .then(() => fail(err, onRejected))
+            .then(() => { throw err })
```

Comment:
@christian-bromann as far as I can see it happens in all cases, I'm not 100% sure, but maybe we can add test scenario's for all the cases we can think of?